testsuite/tests/singlecluster/gateway/reconciliation/change_targetref/test_update_dnspolicy_target_ref.py

```diff
@@ -0,0 +1,43 @@
+"""
+Test for changing targetRef field in DNSPolicy
+"""
+
+from time import sleep
+import pytest
+
+from testsuite.gateway import GatewayListener
+from testsuite.tests.singlecluster.gateway.reconciliation.change_targetref.conftest import change_target_ref
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy]
+
+
+def test_update_dns_policy_target_ref(
+    route, route2, gateway, gateway2, client, client2, dns_policy
+):  # pylint: disable=unused-argument
+    """Test updating the targetRef of DNSPolicy from Gateway 1 to Gateway 2"""
+    assert gateway.wait_until(lambda obj: obj.is_affected_by(dns_policy))
+    assert gateway2.wait_until(lambda obj: not obj.is_affected_by(dns_policy))
+
+    response = client.get("/get")
+    assert not response.has_dns_error()
+    assert response.status_code == 200
+
+    response = client2.get("/get")
+    assert response.has_dns_error()
+
+    dns_ttl = gateway.get_listener_dns_ttl(GatewayListener.name)
+    dns_policy.delete()
```

Comment:
You don't need to delete policy you're testing with here :laughing: What I sent you is just an example how you can extract DNSRecod TTL and then wait for this time after you switch the gateway

Wonder how it didn't fail when I tried to execute it (already know, modify_and_apply uses apply() which creates the object if it doesn't exists, crazy)